R/derive_joined.R

```diff
@@ -0,0 +1,333 @@
+#' Add Variables from a Second Dataset Based on Variables from Both Datasets
+#'
+#' The function adds variables from a second dataset to the input dataset. The
+#' selection of the observations from the second dataset can depend on variables
+#' from both datasets. For example, add the lowest value (nadir) before the
+#' current observation.
+#'
+#' @param dataset Input dataset
+#'
+#'   The variables specified by `by_vars` are expected.
+#'
+#' @param dataset_add Additional dataset
+#'
+#'   The variables specified by the `by_vars`, the `new_vars`, the `join_vars`,
+#'   and the `order` argument are expected.
+#'
+#' @param by_vars Grouping variables
+#'
+#'   The two datasets are joined by the specified variables. Variables from the
+#'   additional dataset can be renamed by naming the element, i.e., `by_vars =
+#'   vars(<name in input dataset> = <name in additinoal dataset>)`.
+#'
+#' @param order Sort order
+#'
+#'   If the argument is set to a non-null value, for each observation of the
+#'   input dataset the first or last observation from the joined dataset is
+#'   selected with respect to the specified order.
+#'
+#'   *Permitted Values*: list of variables or `desc(<variable>)` function calls
+#'   created by `vars()`, e.g., `vars(ADT, desc(AVAL))` or `NULL`
+#'
+#' @param new_vars Variables to add
+#'
+#'   The specified variables from the additional dataset are added to the output
+#'   dataset. Variables can be renamed by naming the element, i.e., `new_vars =
+#'   vars(<new name> = <old name>)`.
+#'
+#'   For example `new_vars = vars(var1, var2)` adds variables `var1` and `var2`
+#'   from `dataset_add` to the input dataset.
+#'
+#'   And `new_vars = vars(var1, new_var2 = old_var2)` takes `var1` and
+#'   `old_var2` from `dataset_add` and adds them to the input dataset renaming
+#'   `old_var2` to `new_var2`.
+#'
+#'   If the argument is not specified or set to `NULL`, all variables from the
+#'   additional dataset (`dataset_add`) are added.
+#'
+#'   *Permitted Values*: list of variables created by `vars()`
+#'
+#' @param join_vars Variables to use from additional dataset
+#'
+#'   The variables required from the additional dataset for `filter_join` should
```

Comment:
"Any extra variables..."?? Instead of "The variables". I think we need to be clear here that you don't need to repeat the `new_vars` variables here

Reply:
Good point, updated.